src/pgwire/src/protocol.rs

```diff
@@ -211,13 +216,55 @@ where
                     .await;
             }
         }
+    } else if use_self_hosted_auth {
+        conn.send(BackendMessage::AuthenticationCleartextPassword)
+            .await?;
+        conn.flush().await?;
+        let password = match conn.recv().await? {
+            Some(FrontendMessage::Password { password }) => password,
+            _ => {
+                return conn
+                    .send(ErrorResponse::fatal(
+                        SqlState::INVALID_AUTHORIZATION_SPECIFICATION,
+                        "expected Password message",
+                    ))
+                    .await
+            }
+        };
```

Comment:
Could potentially consolidate some of these codepaths between frontegg and self managed but probably fine for now